core/src/main/scala/org/broadinstitute/dsde/rawls/resourcebuffer/ResourceBufferService.scala

```diff
@@ -0,0 +1,37 @@
+package org.broadinstitute.dsde.rawls.resourcebuffer
+
+import java.util.UUID
+
+import org.broadinstitute.dsde.rawls.dataaccess.resourcebuffer.ResourceBufferDAO
+import org.broadinstitute.dsde.rawls.model.ProjectPoolType.ProjectPoolType
+import org.broadinstitute.dsde.rawls.model.{GoogleProjectId, PoolId, ProjectPoolId, ProjectPoolType, UserInfo}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ResourceBufferService {
+  def constructor(resourceBufferDAO: ResourceBufferDAO)(userInfo: UserInfo)(implicit executionContext: ExecutionContext): ResourceBufferService = {
+    new ResourceBufferService(resourceBufferDAO, userInfo)
+  }
+}
+class ResourceBufferService(resourceBufferDAO: ResourceBufferDAO, protected val userInfo: UserInfo) {
+
+  def GetGoogleProjectFromRBS(projectPoolType: ProjectPoolType): Future[GoogleProjectId] = getGoogleProjectFromRBS(projectPoolType)
```

Comment:
I know this is the pattern that we use elsewhere in Rawls, but I can't say I remember _why_ we use it? It's always felt unnecessary to me, but maybe that's because I just don't get it

Reply:
this pattern is a remnant of the migration from spray to akka http. We don't need to propagate it in new service classes